fix(client): Only show passwords while depressing the "show" button.

shane-tomlinson opened the pull request:
This builds on #3977 by only showing the password field if the show button is depressed.

shane-tomlinson commented:
cc @ryanfeeley 

Here's the behavior:

![show-pw-on-press](https://cloud.githubusercontent.com/assets/848085/17146904/2c09af3c-5358-11e6-9994-2df283427189.gif)

ryanfeeley commented:
@shane-tomlinson Looks great! If it alleviates issues on change password, should we change the behaviour there?

shane-tomlinson commented:
@ryanfeeley - Thanks! I've added the behavior to both reset password and change password. I'm going to put this up on a dev box so you can play with it and we can try it on mobile. That might take until tomorrow, sometimes dev boxes take a bit of time to set up.

shane-tomlinson commented:
This is running on https://stomlinson.dev.lcip.org. The behavior needs to be updated to correctly handle touch events on mobile devices.

TDA commented:
@shane-tomlinson Is it supposed to be displayed only on click/depress or click/depress+hover? I am seeing some weirdly disconnected behavior:
![stomlinpwdshow](https://cloud.githubusercontent.com/assets/432707/17220688/34353cfa-54a5-11e6-8bf4-5a9d104a272d.gif)

I believe we would need a color scheme change too if we are changing the behavior to not trigger on :hover. The colors are confusing, and a few clicks seem to not be recognized.

shane-tomlinson commented:
> Is it supposed to be displayed only on click/depress or click/depress+hover

Right now it's click/depress and stay within the button.

shane-tomlinson commented:
@ryanfeeley - could you have a look at this behavior on https://stomlinson.dev.lcip.org? I have changed it where the user can depress the button and the show button is only hidden once the user lets up on the button, from anywhere in the page. I have also added touch events.

ryanfeeley commented:
@shane-tomlinson It's hard to reproduce, but I can get it to show and stay shown with a single click.
![click](https://cloud.githubusercontent.com/assets/68213/17248621/c5f7e24a-5569-11e6-9c12-f9b7154e1683.gif)

shane-tomlinson commented:
@philbooth - mind doing a review of this? 

The behavior seen by @ryanfeeley is strange. I am able to reproduce locally, but very carefully. I have to click _and not move the mouse afterwards_. Digging in, it seems the `mouseup` event is not always fired immediately when the element's type changes. Luckily, the problem is short lived, as soon as the mouse is moved, the mouseup event fires and the text is hidden again.

philbooth commented on app/scripts/views/base.js:
```diff
@@ -679,33 +653,72 @@ define(function (require, exports, module) {
-        // IE can blow up if the element is not visible.
+      // Only focus if the device does not have touch. Setting focus
+      // on a touch device causes the virtual keyboard to pop up,
+      // which is jarring.
```
As much as I love reading comments, we have the same comment here repeated just five lines after its original appearance. Is this one surplus to requirements?

shane-tomlinson replied:
Silly me.

philbooth commented:
LGTM, r+. 👍
